src/who_logged.cpp

```diff
@@ -9,59 +10,57 @@
 
-
-    void OnLogin(Player* p)
+    void OnLogin(Player* player)
     {
-        std::string playerIP = p->GetSession()->GetRemoteAddress();
-        std::string playerName = p->GetName();
-        uint32 pAccountID = p->GetSession()->GetAccountId();
-        uint32 pLevel = p->getLevel();
+        std::string playerIP = player->GetSession()->GetRemoteAddress();
+        std::string playerName = player->GetName();
+        uint32 pAccountID = player->GetSession()->GetAccountId();
+        uint32 pLevel = player->getLevel();
         std::string pClass;
         std::ostringstream message;
 
-
-        if (!sConfigMgr->GetBoolDefault("PlayerAnnounce", true))
+        if (!sConfigMgr->GetOption<bool>("PlayerAnnounce", true))
         {
             return;
         }
 
-        switch (p->getClass())
+        switch (player->getClass())
         {
-        case CLASS_WARLOCK:
-            pClass = "Warlock";
-            break;
-        case CLASS_WARRIOR:
-            pClass = "Warrior";
-            break;
-        case CLASS_MAGE:
-            pClass = "Mage";
-            break;
-        case CLASS_SHAMAN:
-            pClass = "Shaman";
-            break;
-        case CLASS_DEATH_KNIGHT:
-            pClass = "Death Knight";
-            break;
-        case CLASS_DRUID:
-            pClass = "Druid";
-            break;
-        case CLASS_HUNTER:
-            pClass = "Hunter";
-            break;
-        case CLASS_PALADIN:
-            pClass = "Paladin";
-            break;
-        case CLASS_ROGUE:
-            pClass = "Rogue";
-            break;
-        case CLASS_PRIEST:
-            pClass = "Priest";
-            break;
+            case CLASS_WARLOCK:
+                pClass = "Warlock";
+                break;
+            case CLASS_WARRIOR:
+                pClass = "Warrior";
+                break;
+            case CLASS_MAGE:
+                pClass = "Mage";
+                break;
+            case CLASS_SHAMAN:
+                pClass = "Shaman";
+                break;
+            case CLASS_DEATH_KNIGHT:
+                pClass = "Death Knight";
+                break;
+            case CLASS_DRUID:
+                pClass = "Druid";
+                break;
+            case CLASS_HUNTER:
+                pClass = "Hunter";
+                break;
+            case CLASS_PALADIN:
+                pClass = "Paladin";
+                break;
+            case CLASS_ROGUE:
+                pClass = "Rogue";
+                break;
+            case CLASS_PRIEST:
+                pClass = "Priest";
+                break;
         }
 
-        sLog->outString("Player '%s' has logged in : Level '%u' : Class '%s' : IP '%s' : AccountID '%u'", playerName.c_str(), pLevel, pClass.c_str(), playerIP.c_str(), pAccountID);
+        LOG_WARN("module", "Player '{}' has logged in : Level '{}' : Class '{}' : IP '{}' : AccountID '{}'", playerName.c_str(), pLevel, pClass.c_str(), playerIP.c_str(), pAccountID);
```

Comment:
why warn? just `LOG_INFO`

Reply:
> why warn? just `LOG_INFO`

Oops, I was testing. Because I had tried with LOG_INFO but it didn't show the information in the console. So, I didn't know if I was doing something wrong, but in the end, it was a log configuration issue I think. The info, I think it does not show them by default, but in reality, the ideal would be to use that one. What happens is that people would not see the message by default.

Reply:
But we can change it to INFO, no problem.

Reply:
Ready. https://github.com/azerothcore/mod-who-logged/pull/7

Reply:
merged!